### PR TITLE
Add oidc-code-flow dev mode test

### LINKED ISSUE #6965: Test that quarkus-oidc works in the dev mode
**Description**
Verify that changing some of the OIDC properties is effective in the dev mode.

### PR DESCRIPTION
**Pull request by sberyozkin**
Fixes #6965.

This test is like the tests in the `oidc-code-flow`. The difference is that it starts with the invalid `client_id` (`client-dev` as opposed to `client-dev-mode` created in Keycloak). As such the exception is expected at the beginning of the test. Next the `client_id` is updated and the test then passes as expected.

### CODE REVIEW

**Comment by sberyozkin** [comment]
It does not run in a native mode which is expected for the devmode tests

**Review by gastaldi — APPROVED** [review]
LGTM, but I'd rather wait for @pedroigor to approve before merging ;)

**Comment by sberyozkin** [comment]
@gastaldi Hi, thanks; sure, lets wait for Pedro and/or Stuart to check

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]
This should really be in oidc/deployment, dev mode tests should not get their own 'integration-tests' module (we already have hundreds of modules, so we need to resist adding any that are not needed).

**Comment by sberyozkin** [comment]
Hi @stuartwdouglas 
The reason I've added it there is that it requires a docker starting KC. I've tried initially allocate it in the `integration/oidc-code-flow` but it was interfering with the main `CodeFlowTest` (KC was failing to create the realms).
If I move it to `extensions/oidc/deployment`, will it be actually tested in CI (it depends on `docker-keycloak` and `test-keycloak` profiles).
CC @gsmet

**Comment by stuartwdouglas** [comment]
Yes, it will still be tested, keycloak is still started for the JVM mode tests.

**Comment by sberyozkin** [comment]
Hi @stuartwdouglas, how does it look now ?

**Comment by sberyozkin** [comment]
`http://localhost:8180` is not available in CI. How to identify the problem ? Looks lie Keycloak has not completed its own initialization yet by the time the test starts contacting Keycloak.

**Comment by sberyozkin** [comment]
@stuartwdouglas The same test works in `integration-tests/`, so I wonder if there is some timing issue.

**Comment by stuartwdouglas** [comment]
It looks like you have fixed the timing issue?

File: extensions/oidc/deployment/pom.xml
Comment (stuartwdouglas) [inline extensions/oidc/deployment/pom.xml]: Failsafe is not used here, as there are no native image tests

File: extensions/oidc/deployment/pom.xml
Comment (stuartwdouglas) [inline extensions/oidc/deployment/pom.xml]: same here

File: extensions/oidc/deployment/pom.xml
Comment (stuartwdouglas) [inline extensions/oidc/deployment/pom.xml]: This should not be here.

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]

**Comment by sberyozkin** [comment]
Hi @stuartwdouglas I've only added a `keycloak.url` maven property and the resource filtering as used in the integration tests, but the main 'fix' was about rebasing and force-pushing repeatedly :-).
Let me take care of the new comments and see what happens :-)

**Comment by stuartwdouglas** [comment]
Does that mean it it likely intermittently failing, and just happened to pass this time?

**Comment by sberyozkin** [comment]
Windows build failure is unrelated: ` Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.1:test (default-test) on project quarkus-resteasy-deployment: There are test failures.`

**Comment by sberyozkin** [comment]
@stuartwdouglas It has passed again, so please merge if all looks good to you

**Comment by sberyozkin** [comment]
@stuartwdouglas It has passed again, so please merge if all looks good to you

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**
